lib/marketplace/app/oauth/index.js

```diff
@@ -0,0 +1,122 @@
+       * @func fetch
+       * @returns {Promise<AppOAuth>}
+       *
+       * @example
+       * import * as contentstack from '@contentstack/management'
+       * const client = contentstack.client({ authtoken: 'TOKEN'})
+       *
+       * client.organization('organization_uid').app('manifest_uid').oauth().fetch()
+       * .then((oAuthConfig) => console.log(oAuthConfig))
+       */
+      this.fetch = async (param = {}) => {
+        try {
+          const headers = {
+            headers: { ...cloneDeep(this.params) },
+            params: {
+              ...cloneDeep(param)
+            }
+          } || {}
+
+          const response = await http.get(this.urlPath, headers)
+          if (response.data) {
+            return response.data.data || {}
+          } else {
+            throw error(response)
+          }
+        } catch (err) {
+          throw error(err)
+        }
+      }
+
+      /**
+       * @description The change oauth details call is used to update the OAuth details, (redirect url and permission scope) of an app.
+       * @memberof Oauth
+       * @func update
+       * @returns {Promise<AppOAuth>}
+       *
+       * @example
+       * import * as contentstack from '@contentstack/management'
+       * const client = contentstack.client({ authtoken: 'TOKEN'})
+       * const config = {
+       *  redirect_uri: 'REDIRECT_URI',
+       *  app_token_config: {
+       *    enabled: true,
+       *    scopes: ['scope1', 'scope2']
+       *   },
+       *   user_token_config: {
+       *    enabled: true,
+       *    scopes: ['scope1', 'scope2']
+       *   }
+       *  }
+       * client.organization('organization_uid').app('manifest_uid').oauth().update({ config })
+       * .then((oAuthConfig) => console.log(oAuthConfig))
+       */
+      this.update = async ({ config, param = {} }) => {
+        try {
+          const headers = {
+            headers: { ...cloneDeep(this.params) },
+            params: {
+              ...cloneDeep(param)
+            }
+          } || {}
+
+          const response = await http.put(this.urlPath, config, headers)
+          if (response.data) {
+            return response.data.data || {}
+          } else {
+            throw error(response)
+          }
+        } catch (err) {
+          throw error(err)
+        }
+      }
+
+      /**
+       * @description The change oauth details call is used to update the OAuth details, (redirect url and permission scope) of an app.
+       * @memberof Oauth
+       * @func getScopes
+       * @returns {Promise<AppOAuth>}
+       *
+       * @example
+       * import * as contentstack from '@contentstack/management'
+       * const client = contentstack.client({ authtoken: 'TOKEN'})
+       *
+       * client.organization('organization_uid').app('manifest_uid').oauth().getScopes()
+       * .then((scopes) => console.log(scopes))
+       */
+      this.getScopes = async () => {
+        try {
+          const headers = {
```

Comment:
Unexpected constant truthiness on the left-hand side of a `||` expression.sonarlint(javascript:S6638)